### ISSUE TITLE
[v2.0.0-beta.1] Invalid SQL statement upon graphql-engine startup when hdb_catalogue schema is present 

### ISSUE THREAD

**Issue by jjbeekman** [hasura/graphql-engine#7070]
When the graphql-engine container spins up and the `hdb_catalogue` schema is already present, the following Postgres error happens.

`2021-06-16 13:20:14.511 UTC [89] ERROR:  argument of CASE/WHEN must be type boolean, not type timestamp with time zone at character 225
2021-06-16 13:20:14.511 UTC [89] STATEMENT:  ALTER TABLE hdb_catalog.event_log ALTER COLUMN locked DROP DEFAULT;
    ALTER TABLE hdb_catalog.event_log ALTER COLUMN locked DROP NOT NULL;
    ALTER TABLE hdb_catalog.event_log ALTER COLUMN locked TYPE TIMESTAMPTZ USING CASE WHEN locked THEN NOW() ELSE NULL END;
 `

How to recreate:
- Take a Postgres database that already contains the `hdb_catalogue` schema, or spin up a database and do a pg_dump
- Spin up the graphql-engine version v2.0.0-beta.1. The error should appear upon startup

When is this issue fixed:
- The graphql-engine (v2) can spin up on a database with existing `hdb_catalogue` schema without failure

**Comment by jjbeekman** [comment]
The SQL statement `ALTER TABLE hdb_catalog.event_log ALTER COLUMN locked TYPE TIMESTAMPTZ USING CASE WHEN locked THEN NOW() ELSE NULL END;` comes from `server/src-rsr/migrations/40_to_41.sql`. It seems that the graphql-engine is always applying the migrations in this folder upon (re)start. Also when the database schema is already present in that state.

**Comment by codingkarthik** [comment]
Hello @jjbeekman,

I was trying to repro the issue following the repro steps you have mentioned in the description of this issue but I was not able to repro it at my side. If you can provide me with the following details, it will be useful for me to debug this issue further:

1. I assume you were running a previous version of graphql-engine on the said DB, can I know which version was it?
2. Did it have any event triggers?
3. Can you run the ```sql select * from hdb_catalog.hdb_version ``` on your DB and share the result here?

Thanks!

**Comment by HosseinYousefi** [comment]
I have the same issue:

**Comment by HosseinYousefi** [comment]
I have the same issue:

```
{"type":"startup","timestamp":"2021-06-28T10:00:53.685+0000","level":"error","detail":{"kind":"catalog_migrate","info":{"internal":{"statement":"-- This migration adds versioning to metadata, used for optimistic locking in UIs.\n-- TODO: Are there changes required in catalog_versions.txt\n\nALTER TABLE hdb_catalog.hdb_metadata ADD COLUMN \"resource_version\" INTEGER NOT NULL DEFAULT 1 UNIQUE;\n","prepared":false,"error":{"exec_status":"FatalError","hint":null,"message":"column \"resource_version\" of relation \"hdb_metadata\" already exists","status_code":"42701","description":null},"arguments":[]},"path":"$","error":"database query error","code":"unexpected"}}
```

1. I used the 2.0.0 beta 1 and now am going to beta 2.
2. I had some event triggers.
3. hasura version from the select query is 43.

**Comment by codingkarthik** [comment]
This is fixed in the v2.0.1 version
